.github/workflows/builder.yml

```
@@ -36,33 +36,36 @@ jobs:
         uses: docker/build-push-action@5cd11c3a4ced054e52742c5fd54dca954e0edd85 # v6.7.0
         with:
           context: ./hassfest
-          build-args: FROM_VERSION=${{ version }}
+          build-args: FROM_VERSION=${{ github.event.inputs.version }}
           load: true
           tags: ${{ env.HASSFEST_IMAGE_TAG }}
 
       # todo test with core
       - name: Checkout core repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
-          name: home-assistant/core
+          repository: home-assistant/core
           path: tmp/core
+          # checkout only the homeassistant folder as others are not needed
+          sparse-checkout: |
+            homeassistant
 
       - name: Run hassfest againt core
-        run: docker run --rm -v ${{ github.workspace }}/tmp/core://github/workspace ${{ env.REGISTRY }}/${{ env.HASSFEST_IMAGE_TAG }}
+        run: docker run --rm -v ${{ github.workspace }}/tmp/core/homeassistant:/homeassistant ${{ env.HASSFEST_IMAGE_TAG }}
```

```suggestion
        run: docker run --rm -v ${{ github.workspace }}/tmp/core/homeassistant/components:/github/workspace ${{ env.HASSFEST_IMAGE_TAG }}
```
Unsure, but I think this is more correct.